control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllersv2
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+)
+
+// MeshGatewayController reconciles a MeshGateway object.
+type MeshGatewayController struct {
+	client.Client
+	Log                  logr.Logger
+	Scheme               *runtime.Scheme
+	MeshConfigController *MeshConfigController
+}
+
+// +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=meshgateway,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=meshgateway/status,verbs=get;update;patch
+
+func (r *MeshGatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// TODO NET-6392 NET-6393 NET-6394 NET-6395
+	return r.MeshConfigController.ReconcileEntry(ctx, r, req, &meshv2beta1.MeshGateway{})
+}
+
+func (r *MeshGatewayController) Logger(name types.NamespacedName) logr.Logger {
+	return r.Log.WithValues("request", name)
```

Comment:
minor (and probably not huge since this is a stub) but should the key here be `"mesh-gateway"` rather than `"request"`?

Reply:
This mimics what happens today for `httproute`, `tcproute`, and `grpcroute`. Gonna leave this as a broader change for a separate PR if that's cool with you @jm96441n 

https://github.com/hashicorp/consul-k8s/blob/f4ccbaa9bc9152722fdb8ff032220f1bc95475e2/control-plane/config-entries/controllersv2/http_route_controller.go#L34

https://github.com/hashicorp/consul-k8s/blob/f4ccbaa9bc9152722fdb8ff032220f1bc95475e2/control-plane/config-entries/controllersv2/tcp_route_controller.go#L34

https://github.com/hashicorp/consul-k8s/blob/f4ccbaa9bc9152722fdb8ff032220f1bc95475e2/control-plane/config-entries/controllersv2/grpc_route_controller.go#L34